### PR TITLE
IGNITE-15602 Add KubernetesConnectionConfiguration.discoveryPort

### PR DESCRIPTION
**Pull request by raymondpwilson**
Add `KubernetesConnectionConfiguration.discoveryPort` so with multiple independent Ignite clusters in a single Kubernetes namespace it is possible to connect to a specific cluster.

### CODE REVIEW

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java [hunk 1]
```diff
@@ -138,6 +141,24 @@ public boolean getIncludeNotReadyAddresses() {
...
 
+    /**
+     * Specifies the port which is returned to the caller to use for service discovery
+     * defaults to 0
+     * @param discoveryPort Port to use for Kubernetes IP Finder
+     * @return {@code this} for chaining.
+     */
+    public KubernetesConnectionConfiguration setDiscoveryPort(int discoveryPort) {
+        this.discoveryPort = discoveryPort;
+        return this;
```
Comment (timoninmaxim) [inline modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:152]: Hi! Could you please align formatting with other public methods:
1. You should put a point in end of lines in all comments.
2. Add additional new line before `@param` and `return`.

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java [hunk 2]
```diff
@@ -138,6 +141,24 @@ public boolean getIncludeNotReadyAddresses() {
         return includeNotReadyAddresses;
     }
 
+    /**
+     * Specifies the port which is returned to the caller to use for service discovery
```
Comment (ptupitsyn) [inline modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:145]:
```suggestion
     * Specifies the port which is returned to the caller to use for service discovery.
```

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java [hunk 3]
```diff
@@ -138,6 +141,24 @@ public boolean getIncludeNotReadyAddresses() {
...
+     * defaults to 0
```
Comment (ptupitsyn) [inline modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:146]:
```suggestion
     * Defaults to 0.
```

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java [hunk 4]
```diff
@@ -138,6 +141,24 @@ public boolean getIncludeNotReadyAddresses() {
...
+     * @param discoveryPort Port to use for Kubernetes IP Finder
```
Comment (ptupitsyn) [inline modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:147]:
```suggestion
     * @param discoveryPort Port to use for Kubernetes IP Finder.
```

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java [hunk 5]
```diff
@@ -138,6 +141,24 @@ public boolean getIncludeNotReadyAddresses() {
...
+     * @return {@code this} for chaining.
+     */
+    public KubernetesConnectionConfiguration setDiscoveryPort(int discoveryPort) {
+        this.discoveryPort = discoveryPort;
```
Comment (ptupitsyn) [inline modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:151]:
```suggestion
        this.discoveryPort = discoveryPort;
        
```

**Review by ptupitsyn — APPROVED** [review]

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java (hunk 1 above)
Comment (raymondpwilson) [reply modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:153]: Done, via Pavel's suggestions.

File: modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java (hunk 4 above)
Comment (raymondpwilson) [reply modules/kubernetes/src/main/java/org/apache/ignite/kubernetes/configuration/KubernetesConnectionConfiguration.java:147]: Have merged all suggestions

**Pull request merged**